runtime/src/lib.rs

```diff
@@ -311,7 +311,7 @@ impl pallet_timestamp::Config for Runtime {
 	type MinimumPeriod = ConstU64<0>;
 	#[cfg(not(feature = "experimental"))]
 	type MinimumPeriod = ConstU64<{ SLOT_DURATION / 2 }>;
-	type WeightInfo = ();
+	type WeightInfo = pallet_timestamp::weights::SubstrateWeight<Runtime>;
```

Comment:
Whilst this is included in the audit report, I am not sure if this is strictly necessary as the weights defined in the pallet are implemented for the unit type (https://github.com/paritytech/polkadot-sdk/blob/c973fe86f8c668462186c95655a58fda04508e9a/substrate/frame/timestamp/src/weights.rs#L84), exactly the same as the implementation used here. It is more explicit however.

I do agree that these should be replaced with actual benchmarks based on this specific runtime configuration on reference hardware, which is not the case when using the weights from the frame pallets as that benchmarked runtime _may_ have a different config to this one.

Reply:
Ah good catch, I missed this.

Same story with `pallet_sudo` - https://github.com/paritytech/polkadot-sdk/blob/c973fe86f8c668462186c95655a58fda04508e9a/substrate/frame/sudo/src/weights.rs#L108

Reply:
Every `WeigthInfo` implementation for each pallet has at the moment of writing the same weights defined than those in their `SubstrateWeight` implementation.

While the implementation for `()` is meant to exist for testing and the one provided in `SubstrateWeight` is meant to follow Polkadot's hw requirements, they seem to be updated together.

`()` does a better job making the developer aware that `WeightInfo` should be updated with their benchmark results.